test/xbgpu/test_engine.py

```diff
@@ -1014,9 +1015,9 @@ def heap_factory(batch_index: int) -> list[spead2.send.HeapReference]:
         # To keep it manageable, compare a batch at a time.
         for i in range(expected_beams.shape[0]):
             # NOTE: As per the explanation at the end of `_send_data`, we
-            # only verify up the the penultimate heap's data for each `beam_result`
-            # because the final heap is sent coincidentally - not because it is
-            # expected to have sane data in it.
+            # only verify data in the range of `batch_indices` for each
+            # `beam_result` as any heaps sent afterwards are sent by default
+            # - not because it is expected to have sane data in it.
```

Comment:
You've switched from plural ("heaps .. are") to singular ("it is") mid-sentence.